### ISSUE TITLE
Wildcards not working at all (Windows Portable Edition)

### ISSUE THREAD

**Issue by bobpuffer1** [adieyal/comfyui-dynamicprompts#10]
Sample Random right out of the chute
get 16 green balls when batch size set to 16
get 16 red the next time it's run
then 16 blue
error on console "WARNING:dynamicprompts.samplers.utils:No values found for wildcard artists/Photography/by_country/georgia" though I have the wildcards files in both the ComfyUI/wildcards and ComfyUI/custom_nodes/comfyui-dynamicprompts/wildcards
number of batches doesn't change behavior

Sample Combinatorial right out of the chute 
always get 64 red balls no matter how many times I run it when batch size set to 64
get same warning in console as above
batch number doesn't affect behavior

**Comment by bobpuffer1** [comment]
Some further research:

- It didn't find the wildcard files because they were nested one directory below where it expected them to be. Cloning from github uncovered that. 
- It still doesn't randomize inside of a batch. If I set batch members to 8 I get 8 of the same actress when __actress__ is the only wildcard in the prompt. 
- The only way I could get it to randomize is to set batch members to 1, number of batches to 1 and have it reprompt every time the queue hit zero. You then have to physically uncheck the box for reprompting to end the generations.

**Comment by adieyal** [comment]
Could you please test the latest version? It now looks for wildcards in custom_nodes/comfyui-dynamicprompts

**Comment by adieyal** [comment]
Also test to make sure it randomizes in between batches. I'm not sure how to randomize within a batch since the prompt is generated earlier in the chain

**Comment by bobpuffer1** [comment]
yeah it sees the wild card files
it doesn't randomize inside of a batch and I think that's appropriate behavior the problem is that comfy UI doesn't observe the number of batches that you tell it to reprompt once the queue gets to zero and then it runs forever
so that is a comfy ui problem i'm going to list it as an issue with their github
otherwise it looks like dynamic prompts are working appropriately I havent spent much time on combinatorial because it doesn't really work very well for me being it starts over between sessions

**Comment by adieyal** [comment]
Re combinatorial. Do you mean that it starts over when you restart ComfyUI? There is a setting shuffles the order of the prompts in combinatorial. I haven't exposed it yet though.

**Comment by bobpuffer1** [comment]
- I used combinatorial example from your README modifying the country to Hungary since therre's only one photographer in Georgia.. There are thirteen in Hungary.
- it iterates 13 times through the batch in order to change colors because there's 13 artists in the wildcards file
- there was never a point where red balls and green balls shared the same batch
- There appeared to  never be a time when different artists shared the same batch
- in order to accomplish what I considered be the primary use case scenario I set the batch members to one and the batch count under extra options to one Unchecked automatically reprompt when Q gets to zero.
- This creates 39 images one for each combination of artist/ ball color
- I believe the batch works as I would anticipate it I would expect that inside of the batch it would all be one color one artist.  Only be between batches that the ball color and the artists would change.
- the hitch in the Gidyup is that comfyui handles batch count incorrectly at least for dynamic prompts
- as in random prompts it should reprompt after each batch not just when the queue gets to zero
- I've started a discussion on comfyUI Githup About this apparent issue
- In the meantime the workaround is to set the batch count to one have it reprompt when the queue gets to 0 and manually turn it off when all the batches are completed. 
- This is less of a problem because comfy UI is so fast if we were generating overnight this would be a real problem
- incidentally if it isn't clear from the notes above dynamic prompts does cycle the wild cards between generations or batches. it would be good to have that setting exposed no I might never use it.
- I couldn't tell but if dynamic prompts is still choosing its wildcards alphabetically that's a problem

**Comment by bobpuffer1** [comment]
I hate having to revisit this issue but I have to since I am forced to go back to Windows stand alone edition of comfyUI.
I initially thought that the windows standalone edition was installed incorrectly because it placed everything one directory below where it was expected to operate. but I determined that it was installed correctly the way that it has to be installed in order for all of the batch files to work correctly. 

the problem is that dynamic prompts looks for the wildcards folder in relation to the home directory and wild cards folder is always one step below where it's expected to be when it's in the windows stand alone edition. I proved this concept by placing the wild cards folder One directory up from where it logically should have been and dynamic prompts began working again. this was an easy workaround but isn't really a sustainable implementation. perhaps dynamic prompts needs to look for a wild cards in relation to the comfyUI directory I don't know what the right answer is but I know that I had to go to the windows stand alone edition because WAS custom nodes only supports the windows stand alone edition.

**Comment by karl-von-knecht** [comment]
I was able to make wildcard text files work by putting them in C:\ComfyUI_windows_portable\ComfyUI\wildcards\
This is different from what the installation instructions say, so it would be good to update those.

**Comment by bobpuffer1** [comment]
Yes, I agree. the wild cards files work exactly as you've described. The problem is that they also work if they're placed in the custom._ nodes, directory, but that. forces the startup to consider them to be a custom node, and they fail that test.. The much more. important issue because this one can actually be worked around. Is that they don't update unless you restart comfyUI. adding new wildcards file or changing wildcards files' contents isn't ever reflected until you restart Comfy UI..

**Comment by rafstahelin** [comment]
> Yes, I agree. the wild cards files work exactly as you've described. The problem is that they also work if they're placed in the custom._ nodes, directory, but that. forces the startup to consider them to be a custom node, and they fail that test.. The much more. important issue because this one can actually be worked around. Is that they don't update unless you restart comfyUI. adding new wildcards file or changing wildcards files' contents isn't ever reflected until you restart Comfy UI..

did you find a way to avoid this issue? 

Also I dont get changes with  {red|green|blue} ball. It's always a red ball
Is there a reason?

Wildcards do work for me but annoying not to be able to change them and have to restart to take effect
